wicket-core/src/main/java/org/apache/wicket/Application.java

```diff
@@ -1569,23 +1569,12 @@ public final RequestCycle createRequestCycle(final Request request, final Respon
 			@Override
 			public void onDetach(final RequestCycle requestCycle)
 			{
-				IPageManager pageManager;
-				
-				if (Session.exists())
-				{
-					pageManager = Session.get().getPageManager();
-				} else {
-					pageManager = internalGetPageManager();
-				}
-				pageManager.detach();
+				internalGetPageManager().detach();
```

Comment:
Sufficient to detach the page manager only. No need to detach the session here, RequestCycle#onDetach() does it already.